### PR TITLE
Implement Quick Builds with Smashables

### LINKED ISSUE #60: Specific smashable doesn't have a respawn cooldown
Specific smashable doesn't have a respawn cooldown. I also think I've seen this same thing happen to one of the crates at Sentinel Point Zeta at CP
[Video](https://youtu.be/m3jDAhI0g8gl)

### PR DESCRIPTION
**Pull request by TheNexusAvenger**
Closes https://github.com/UchuServer/Uchu/issues/60

Some smashables in Lego Universe expose a quick build when smashed, and the smashable respawns after the quick build expires. This pull request implements this behavior instead of the existing behavior where the smashable instantly respawns and the quick build is always visible.

### CODE REVIEW

**Review by MickVermeulen — APPROVED** [review]
Nice change! 😄

**Review by Jettford — CHANGES_REQUESTED** [review]
How this is setup needs to be changed before it can be considered for a merge

**Comment by MickVermeulen** [comment]
@Jettford could you be specific on what should be changed in your opinion?

**Comment by Jettford** [comment]
This can all be done in one component

**Comment by Jettford** [comment]
The LDF key spawnNetNameForSpawnGroupOnSmash contains the information for the quick build to find its parent.

**Comment by MickVermeulen** [comment]
Isn't that what's happening on line 132 in the quick build component? ` if (GameObject.Settings.TryGetValue("spawnNetNameForSpawnGroupOnSmash",out var destructibleSpawnerName))`

**Comment by Jettford** [comment]
Yeah but it is really spread out for some reason.

**Comment by Jettford** [comment]
I mean it works, but it just feels bodge

**Comment by TheNexusAvenger** [comment]
I agree it feels like a bodge. Communicating between components like this didn't feel like the proper solution, but I couldn't figure out anything else reasonable since the quick build knows of the spawner, and I wasn't sure how to go from there. It isn't like I had a reference, like OpCrux, since it doesn't appear to be implemented in the public projects.

**Comment by Jettford** [comment]
You shouldn't really need reference from other servers.

**Comment by MickVermeulen** [comment]
I think we can probably merge this if we don't have any other short term solutions. Maybe leave a comment there with a TODO on a possible improvement.

**Comment by MickVermeulen** [comment]
Okay so I looked into it more in-depth and I also can't think of a scenario where all the code is either in the quick build or spawner component. I do agree that this kind of direct connections between Components is not preferred but I don't see any other solution. Jett how did you envision this being done in one component? The only thing I came up with is not storing the QuickBuildComponent on the SpawnerComponent but passing a function to the spawner component that would execute the logic provided by the QuickBuildComponent. That way the logic is still distributed but at least they don't directly reference each other.

File: Uchu.World/Objects/Components/ReplicaComponents/QuickBuildComponent.cs
Comment (MickVermeulen) [inline Uchu.World/Objects/Components/ReplicaComponents/QuickBuildComponent.cs]: Didn't you mean to do `gameObject` here? Now the parent object of the quick build component is being hidden.

Comment (TheNexusAvenger) [reply Uchu.World/Objects/Components/ReplicaComponents/QuickBuildComponent.cs]: It is intended. That code hides the initial quick build, which gets made visible when the spawner's object is destroyed.

Comment (MickVermeulen) [reply Uchu.World/Objects/Components/ReplicaComponents/QuickBuildComponent.cs]: Thanks for clarifying!

**Comment by MickVermeulen** [comment]
I implemented a change that only utilizes events, note that we have to wait for the refactor to be merged as I merged it into this branch which in hindsight was a stupid idea but otherwise the refactor is merged on accident.

**Comment by TheNexusAvenger** [comment]
Tried out the new implementation, it doesn't act correctly in a bunch of cases I tried, such as the pirate gate in Gnarled Forest and the hidden bouncers in Forbidden Valley. The video demos cover the current problems:
https://imgur.com/RhXoNbQ
https://imgur.com/088zXl5

**Pull request merged**
